### PR TITLE
Issue 1 optional call args

### PR DESCRIPTION
**Pull request by xsamix**

### CODE REVIEW

**Comment by xsamix** [comment]
I didn't commit the builded js yet, as I dunno if there's differences on the results between versions of go & gopherjs (I have go-1.7 on my machine).

Btw, I thought about checking for arg types from call method so you can just leave onLog (js.Object) away and just define address (string). But, not really sure if that's ok with Go, as args are typed after all. So what I mean with this is, I'd check if onLog is string instead of a function, and if so, assign onLog content to address. Could implement this in js but don't know if that's ok with go and gopherjs.

**Comment by plokk** [comment]
If I recall correctly, at least onLog should've been optional already, but actually wasn't. I ran into this while making bot example in june. So in that sense this is a good improvement. Can't say anything code-wise tho.

File: src/ninchatclient/lib/call.go [hunk 1]
```diff
@@ -6,6 +6,10 @@ import (
 )
 
 func call(params map[string]interface{}, onLog *js.Object, address string) *js.Object {
```
Comment (tsavola) [inline src/ninchatclient/lib/call.go]: The correct way to detect undefined JavaScript value is to change the Go function signature to
```
func call(params map[string]interface{}, onLog *js.Object, address *js.Object) *js.Object {
```
and then check manually if `address` is `js.Undefined`. Go string may then be extracted with `address.String()`.

**Review by tsavola — CHANGES_REQUESTED** [review]

**Comment by xsamix** [comment]
@tsavola ping

File: src/ninchatclient/lib/call.go [hunk 2]
```diff
@@ -5,9 +5,11 @@ import (
 	"github.com/ninchat/ninchat-go"
 )
 
-func call(params map[string]interface{}, onLog *js.Object, address string) *js.Object {
-	if address == "undefined" {
-		address = ""
+func call(params map[string]interface{}, onLog *js.Object, address *js.Object) *js.Object {
+	var api string
```
Comment (tsavola) [inline src/ninchatclient/lib/call.go]: Weird variable name choice...

**Review by tsavola — APPROVED** [review]

**Comment by xsamix** [comment]
@tsavola pong

File: src/ninchatclient/lib/call.go [hunk 3]
```diff
@@ -6,10 +6,10 @@ import (
 )
 
 func call(params map[string]interface{}, onLog *js.Object, address *js.Object) *js.Object {
-	var api string
+	var apihost string
```
Comment (tsavola) [inline src/ninchatclient/lib/call.go]: Ah, now I get (also the original) name! :)

**Review by tsavola — APPROVED** [review]

**Pull request merged**